src/widget/whighlightinggroup.h

```diff
@@ -0,0 +1,59 @@
+#ifndef WHIGHLIGHTINGGROUP_H
+#define WHIGHLIGHTINGGROUP_H
+
+#include "widget/wwidgetgroup.h"
+
+class SkinContext;
+
+// The WHighlightingGroup can be used to highlight a group of widgets by
+// changing the background or any other css style option.
+// Example skin:
+// <HighlightingGroup>
+//  <Id>1</Id>
+//  <BackPathHighlighted>style/style_bg_effect1_high</BackPathHighlighted>
+//  <Connection>
+//   <ConfigKey>[EffectRack1_EffectUnit1],single_effect_focus</ConfigKey>
+//   <BindProperty>highlightedId</BindProperty>
+//  </Connection>
+// </HighlightingGroup>
+
+class WHighlightingGroup : public WWidgetGroup {
+    Q_OBJECT
+  public:
+    explicit WHighlightingGroup(QWidget* pParent = nullptr);
+
+    // The highlight property is used to restyle the widget with CSS.
+    // The declaration #MyGroup[highlight="1"] { } will define the style
+    // for the highlighted state. Note: The background property does not
+    // support color schemes for images, a workaround is to set the background
+    // image via <BackPath> and <BackPathHighlighted> from the skin.
+    Q_PROPERTY(int highlight READ getHighlight WRITE setHighlight NOTIFY highlightChanged)
+
+    // The highlightedId property is used to select the highlighted widget
+    // group by its Id, which can be set from skin by <Id>4</Id>
+    Q_PROPERTY(int highlightedId READ getHighlight WRITE setHighlightedId NOTIFY highlightChanged)
```

Comment:
Yes, that provides more flexibility, but to me it does not really fit the idea of selecting one member of a group. What do we really want to provide with this? A general way for skins to apply different styles in response to changes in COs? I like that idea, but I don't think the concept of a `highlightedId` makes much sense with it. An ID implies that it is one member of a group. Perhaps we should have a general way for a QSS-accessible property to mirror the state of a CO. For example:

in skin XML:
```
<WidgetGroup>
  <ObjectName>ExampleWidgetGroup</ObjectName>
  <Children>
    whatever widgets here
  </Children>
</WidgetGroup>
<Connection>
  <Connection>[Channel1],play</Connection>
  <BindProperty>connection</BindProperty>
</Connection>
```

in skin QSS:
```
#ExampleWidgetGroup[connection="1"] {
  style to apply when [Channel1],play is 1
}
```

This will allow different styles to be applied for each state of a CO, not just 0 and 1.